x-pack/plugins/asset_manager/public/plugin.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CoreSetup, CoreStart, PluginInitializerContext } from '@kbn/core/public';
+import { Logger } from '@kbn/logging';
+import { AssetManagerPluginClass } from './types';
+import { PublicAssetsClient } from './lib/public_assets_client';
+import type { AssetManagerConfig } from '../common/config';
+
+export class Plugin implements AssetManagerPluginClass {
+  public config: AssetManagerConfig;
+  public logger: Logger;
+
+  constructor(context: PluginInitializerContext<{}>) {
+    this.config = context.config.get();
+    this.logger = context.logger.get();
+  }
+
+  setup(core: CoreSetup) {
+    // Check for config value and bail out if not "alpha-enabled"
+    if (!this.config.alphaEnabled) {
+      this.logger.info('Public is NOT enabled');
+      return;
+    }
+
+    this.logger.info('Public is enabled');
+
+    const publicAssetsClient = new PublicAssetsClient(core.http);
+    return {
+      publicAssetsClient,
+    };
+  }
+
+  start(core: CoreStart) {
+    // Check for config value and bail out if not "alpha-enabled"
+    if (!this.config.alphaEnabled) {
+      return;
+    }
+
+    const publicAssetsClient = new PublicAssetsClient(core.http);
+    return {
```

Comment:
Is there a type to be used by other plugins for this interface?  Something like `AssetManagerClientStartExports`

Reply:
Yeah, it's exported from index -> https://github.com/elastic/kibana/pull/167191/files/a56171afa68caebebf44a4aec662060f51e6f2e4#diff-7a9e97b54f3d0bed47224ea2683d775cfd0487caeda0707e6ce865259e4f264aR19